src/ensembl/src/shared/accordion/Accordion/Accordion.wrapper.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+import { UUID } from '../ItemContainer/ItemContainer';
+
+import { Consumer, Provider } from '../AccordionContainer/AccordionContainer';
+import Accordion from './Accordion';
+import accordionStyles from '../styles/Accordion.scss';
+
+type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
+
+type AccordionWrapperProps = Omit<
+  React.HTMLAttributes<HTMLDivElement>,
+  'onChange'
+> & {
+  accordion?: boolean;
+  onChange(args: UUID | UUID[]): void;
+};
+
+export default class AccordionWrapper extends React.Component<
```

Comment:
Any reason why you didn't like using a functional component with hooks? Just curious 😄

Reply:
This is something I wanted to have a discussion about. At the moment, I just used the code from the package `react-accessible-accordian` which is based on classes. Before I start to re-write it to use functional components, I would like to know if we are not planning to use class components for any particular reason?

Reply:
We might use class components when it is difficult to write a component with functions. However, I think we will mainly stick with functional components with hooks.